sdl.cpp

```diff
@@ -0,0 +1,294 @@
+#include <string>
+#include <SDL.h>
+#include <SDL_image.h>
+#include <SWI-cpp2.h>
+
+PREDICATE(sdl_init_, 1)
+{
+  if (SDL_Init(A1.as_uint32_t()) < 0)
+  {
+    throw PlException(PlTerm_string(SDL_GetError()));
+  }
+  return true;
+}
+
+PREDICATE(sdl_quit, 0)
+{
+  SDL_Quit();
+  return true;
+}
+
+PREDICATE(sdl_createwindow_, 7)
+{
+  SDL_Window *window = SDL_CreateWindow(A2.as_string().c_str(),
+                                        A3.as_int(), A4.as_int(),
+                                        A5.as_int(), A6.as_int(),
+                                        A7.as_uint32_t());
+  if (window == NULL)
+  {
+    throw PlException(PlTerm_string(SDL_GetError()));
+  }
+  return A1.unify_pointer(window);
+}
+
+PREDICATE(sdl_destroywindow, 1)
+{
+  SDL_DestroyWindow((SDL_Window *) A1.as_pointer());
+  return true;
+}
+
+PREDICATE(sdl_createrenderer_, 4)
+{
+  SDL_Renderer *renderer = SDL_CreateRenderer((SDL_Window *) A2.as_pointer(),
+                                              A3.as_int(),
+                                              A4.as_uint32_t());
+  if (renderer == NULL)
```

Comment:
Usually people write `if (!renderer)`
Otherwise, C++ replaces `NULL` with `nullptr`

Reply:
>  Otherwise, C++ replaces NULL with nullptr

So, I'm using a `NULL` comparison here because the SDL documentation explicitly says that it will return `NULL` when encountering an error: https://wiki.libsdl.org/SDL2/SDL_CreateRenderer#return-value